### ISSUE TITLE
Better automatic inbetweening / stroke matching

### ISSUE THREAD

**Issue by dalboris** [dalboris/vpaint#47]
Users should be able to select any selection of cells at frame 1, and any selection of cells at frame 2, then hit "automatic inbetween" to generate all required inbetween cells to interpolate frame 1 into frame 2.

Obviously, this is a very hard research problem, but I'll see how far I can go down that line.

See also #40 for an alternative UI as implemented in Cacani.

**Comment by blurymind** [comment]
I think cacani does this beautifully by showing in the onion skin the order of line drawing and the direction of line drawing.

**Comment by dalboris** [comment]
Yes, but unfortunately it is limited as it doesn't allow you to change topology over time, which is the whole point of VPaint ;-) What I have in mind is much more useful and general.

**Comment by blurymind** [comment]
here is some inspiration :)
http://www.cgsociety.org/news/article/1404/autocomplete-hand-drawn-animations

**Comment by dalboris** [comment]
Wow, this is awesome, thx for sharing! I didn't have a look at Siggraph Asia papers yet, and I will definitely have a careful look at the math they use, I might borrow some. Obviously, the UI is good inspiration too

Our case is a bit different though since we want to have connectivity information which they don't seem to have. [This work](https://graphics.ethz.ch/publications/papers/paperWhi10.php) is closer to what I have in mind, but I should totally take inspiration from both, and combine them in the context of the data structure I have. 

Very exciting, I'm glad other smart people are looking into this problem! :-)

**Comment by blurymind** [comment]
this is a very interesting paper! Thank you for sharing. :)
I hope that many of these features find their way to vpaint.

**Comment by blurymind** [comment]
vpain still has no way of achieving arcs in its inbetweening!
See point 3.4 of their paper:
https://graphics.ethz.ch/Downloads/Publications/Papers/2010/Whi10/Whi10.pdf

I think thats a pretty important one too. Cacani can do it already. Not sure about tween maker, I think they do it by using joints

**Comment by dalboris** [comment]
Yes, I am well aware of this, this will come. :)

**Comment by dalboris** [comment]
Not that I had already implemented most of this paper in the past, including the logarithmic spirals.

**Comment by blurymind** [comment]
ah i see.. then looking forward to that github commit that brings those goodies to vpaint :P

**Comment by blurymind** [comment]
btw blender devs are getting interested in vpaint. Aligorith in perticular is the main dev of blender's grease pencil and has also worked on the animation system.
http://blenderartists.org/forum/showthread.php?384123-Autocomplete-hand-drawn-animations-SIGGRAPH-Asia-2015

**Comment by dalboris** [comment]
Great to hear!
